sound/soc/codecs/rt700.c

```diff
@@ -358,12 +351,27 @@ static int rt700_set_jack_detect(struct snd_soc_component *component,
 		regmap_write(rt700->regmap,
 			RT700_SET_AUDIO_POWER_STATE, AC_PWRST_D3);
 
+	dev_dbg(&rt700->slave->dev, "in %s done\n", __func__);
+	mod_delayed_work(system_power_efficient_wq,
+			&rt700->jack_detect_work, msecs_to_jiffies(250));
+
+}
+
+static int rt700_set_jack_detect(struct snd_soc_component *component,
+	struct snd_soc_jack *hs_jack, void *data)
+{
+	struct rt700_priv *rt700 = snd_soc_component_get_drvdata(component);
+
```

Comment:
It's not clear to me what happens if we were to use set_jack(component, NULL, NULL), as done in some machine drivers. I think it's a way to 'deactivate' the jack for suspend/resume, but I am not clear on this. @bardliao can you comment?

Reply:
@plbossart @shumingfan Yes, codec driver should disable jack detection when set_jack(component, NULL, NULL) is called. That's how we disable/enable jack detection.

Reply:
@bardliao I will update the patch today.